### PR TITLE
Add data grid sorting

### PR DESCRIPTION
**Pull request by vinothvino42**
Ticket: #127

Sample YAML:
```yaml
View:
  header:
    title: DataGrid
  styles:
    scrollableView: true
  onLoad:
    invokeAPI:
      name: getHotels

  body:
    Column:
      styles:
        gap: 16
        padding: 24
      children:
        - DataGrid:
            id: dataGridId
            horizontalMargin: 20
            dataRowHeight: 50
            headingRowHeight: 80
            columnSpacing: 20
            dividerThickness: 1
            sorting:
              columnIndex: 0
              order: descending
            DataColumns:
              - label: Hotel Name
                type: text
                tooltip: Hotel Name
                sortable: true
                sortKey: name

              - label: Location
                type: text
                tooltip: Location
                sortable: true
                sortKey: location

              - label: Price
                type: number
                tooltip: Price
                sortable: false
                sortKey: price

            item-template:
              data: ${getHotels.body.hotels}
              name: hotel
              template:
                DataRow:
                  children:
                    - Text:
                        text: ${hotel.name}
                    - Text:
                        style: { font: subtitle }
                        text: ${hotel.location}
                    - Text:
                        text: ${hotel.price}

API:
  getHotels:
    authorization: none
    uri: "https://pastebin.com/raw/N460wNj5"
    method: "GET"

```

### CODE REVIEW

File: lib/layout/data_grid.dart [hunk 1]
```diff
@@ -200,20 +203,15 @@ class DataGridController extends BoxController {
 class DataColumnSort {
   int? columnIndex;
```
Comment (vusters) [inline lib/layout/data_grid.dart:204]: This is required, otherwise we don't have a valid sort object.

File: lib/layout/data_grid.dart [hunk 2]
```diff
@@ -251,9 +250,7 @@ class DataGridState extends WidgetState<DataGrid> with TemplatedWidgetState {
           onDataChanged: (List dataList) {
         this.dataList = dataList;
         print('RegisterItemTemplate called');
```
Comment (vusters) [inline lib/layout/data_grid.dart:252]: remove

**Review by vusters — APPROVED** [review]

**Comment by vusters** [comment]
Go ahead and merge this. Pleaes make sure schema is up to date.

File: lib/layout/data_grid.dart (hunk 2 above)
Comment (vinothvino42) [reply lib/layout/data_grid.dart]: Removed

**Comment by vinothvino42** [comment]
> Go ahead and merge this. Pleaes make sure schema is up to date.

Updated the schema

**Pull request merged**